crates/transport-ipc/src/mock.rs

```diff
@@ -43,47 +120,152 @@ impl Default for MockIpcServer {
 }
 
 impl MockIpcServer {
-    /// Create a new mock IPC server.
+    /// Create a new mock IPC server instance.
+    /// This creates a temporary file to use as the Unix domain socket path.
     pub fn new() -> Self {
-        Self { replies: VecDeque::new(), path: NamedTempFile::new().unwrap() }
-    }
+        let temp_file = Arc::new(NamedTempFile::new().expect("Failed to create temp file"));
+        let path = temp_file.path();
 
-    /// Add a raw reply to the server.
-    pub fn add_raw_reply(&mut self, reply: Vec<u8>) {
-        self.replies.push_back(reply);
+        // Clean up any existing socket file at this path
+        if path.exists() {
+            let _ = fs::remove_file(path);
+        }
+
+        Self {
+            replies: Arc::new(Mutex::new(VecDeque::new())),
+            temp_file,
+            shutdown: Arc::new(Mutex::new(None)),
+        }
     }
 
-    /// Add a reply to the server.
-    pub fn add_reply<S: Serialize>(&mut self, s: S) {
-        let reply = serde_json::to_vec(&s).unwrap();
-        self.add_raw_reply(reply);
+    /// Get the path to the Unix domain socket that this server will listen on.
+    /// This path should be passed to the provider being tested.
+    pub fn path(&self) -> PathBuf {
+        self.temp_file.path().to_owned()
     }
 
-    /// Add a json-rpc response to the server.
-    pub fn add_response<S: Serialize>(&mut self, response: Response<S>) {
-        self.add_reply(response);
+    /// Create a new handle to control this server.
+    /// The handle can be used to add responses and trigger shutdown.
+    pub fn handle(&self) -> MockIpcHandle {
+        MockIpcHandle {
+            replies: self.replies.clone(),
+            shutdown: self.shutdown.clone(),
+            _temp_file: self.temp_file.clone(),
+        }
     }
 
-    /// Get the path to the socket.
-    pub fn path(&self) -> PathBuf {
-        self.path.path().to_owned()
+    /// Handle a single client connection.
+    /// This function runs in a separate task for each connected client.
+    async fn handle_connection(
+        mut stream: tokio::net::UnixStream,
+        replies: Arc<Mutex<VecDeque<Vec<u8>>>>,
+    ) {
```

Comment:
I assume we could reuse the ReadJsonStream here

https://github.com/alloy-rs/alloy/blob/c475390db3a55df62a4ecf2ab1134e55a3d0eb3d/crates/transport-ipc/src/lib.rs#L105-L105

Reply:
Yep, it does indeed seem like we can, addressed.